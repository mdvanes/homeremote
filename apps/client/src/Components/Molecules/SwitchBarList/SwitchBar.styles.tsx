import { makeStyles, Theme, StyleRules, createStyles } from "@mui/material";

const useStyles = makeStyles(
    ({ spacing, typography }: Theme): StyleRules =>
        createStyles({
            card: {
                marginBottom: spacing(1),
                "> div": {
                    paddingBottom: "0 !important",
                },
            },
            buttonGroup: {
                display: "flex",
                padding: "0 !important",
                alignItems: "center",
                "& button": {
                    backgroundColor: "transparent",
                    border: "none",
                    cursor: "pointer",
                    padding: "0.7em 1.2em",
                    fontSize: typography.body1.fontSize,
                },
                "& .dummy": {
                    height: spacing(3),
                    width: spacing(3),
                },
            },
            label: {
                fontFamily: "Roboto",
                flex: 1,
                textAlign: "center",
                "& button": {
                    fontFamily: "Roboto",
                },
            },
        })
);

export default useStyles;
