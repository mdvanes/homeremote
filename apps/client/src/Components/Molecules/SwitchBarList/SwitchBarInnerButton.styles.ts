import { makeStyles } from "tss-react/mui";

const useStyles = makeStyles()(({ palette }) => ({
    root: {
        color: palette.primary.light,
        "&:hover": {
            color: palette.background.paper,
            backgroundColor: palette.primary.light,
        },
    },
    active: {
        backgroundColor: `${palette.primary.light} !important`,
        color: palette.background.paper,
    },
}));

export default useStyles;
