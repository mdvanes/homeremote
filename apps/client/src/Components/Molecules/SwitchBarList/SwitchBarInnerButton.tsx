import React, { FC } from "react";
import { Icon } from "@mui/material";
import useStyles from "./SwitchBarInnerButton.styles";

type Props = {
    isReadOnly: boolean;
    clickAction: (
        event: React.MouseEvent<HTMLButtonElement, MouseEvent>
    ) => void;
    icon: string;
    isActive: boolean;
};

const SwitchBarInnerButton: FC<Props> = ({
    isReadOnly,
    clickAction,
    icon,
    isActive,
}) => {
    const { classes } = useStyles();
    return (
        <button
            onClick={isReadOnly ? undefined : clickAction}
            className={isActive ? classes.active : classes.root}
        >
            {isReadOnly ? (
                <div className="dummy"></div>
            ) : (
                <Icon className="material-icons">{icon}</Icon>
            )}
        </button>
    );
};

export default SwitchBarInnerButton;
