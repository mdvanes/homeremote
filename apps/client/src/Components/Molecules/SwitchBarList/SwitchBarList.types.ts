// Types should be in @homeremote/types/switches.types.ts but it errors with: Support for the experimental syntax 'flow' isn't currently enabled
export const DOMOTICZ_SELECTOR_STATES_OPTIONS = {
    0: "disconnected",
    10: "disarmed",
    20: "partarmed",
    30: "armed",
} as const;

type DomoticzSelectorStateCodes = keyof typeof DOMOTICZ_SELECTOR_STATES_OPTIONS;

const DOMOTICZ_SELECTOR_STATECODE_OPTIONS: DomoticzSelectorStateCodes[] = [
    0, 10, 20, 30,
];

export const isDomoticzSelectorStateCode = (
    nr: number | null
): nr is DomoticzSelectorStateCodes => {
    return DOMOTICZ_SELECTOR_STATECODE_OPTIONS.some((option) => option === nr);
};
