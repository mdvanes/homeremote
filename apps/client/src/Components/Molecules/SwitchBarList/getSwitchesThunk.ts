import { SwitchesResponse } from "@homeremote/types";
import { createAsyncThunk } from "@reduxjs/toolkit";
import fetchToJson from "../../../fetchToJson";

// Extracted from switchBarListSlice to be able to mock it
export const getSwitches = createAsyncThunk<SwitchesResponse>(
    `switchesList/getSwitches`,
    async () => fetchToJson<SwitchesResponse>("/api/switches")
);
