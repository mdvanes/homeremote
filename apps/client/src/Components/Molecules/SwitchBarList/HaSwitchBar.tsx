import { HomeRemoteHaSwitch } from "@homeremote/types";
import { FC } from "react";
import SwitchBar from "./SwitchBar";
import SwitchBarInnerButton from "./SwitchBarInnerButton";

interface HaSwitchBarProps {
    haSwitch: HomeRemoteHaSwitch;
}

const HaSwitchBar: FC<HaSwitchBarProps> = ({ haSwitch }) => {
    // Implement the HaSwitchBar component logic here

    return (
        <SwitchBar
            label={haSwitch.name}
            leftButton={
                <SwitchBarInnerButton
                    isReadOnly={false}
                    clickAction={() => {
                        /* */
                    }}
                    icon="radio_button_checked"
                    isActive={haSwitch.status === "On"}
                />
            }
            rightButton={
                <SwitchBarInnerButton
                    isReadOnly={false}
                    clickAction={() => {
                        /* */
                    }}
                    icon="radio_button_unchecked"
                    isActive={haSwitch.status === "Off"}
                />
            }
            icon={false}
            labelAction={false}
        />
    );
};

export default HaSwitchBar;
