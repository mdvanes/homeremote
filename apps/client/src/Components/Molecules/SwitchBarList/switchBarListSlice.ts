import { HomeRemoteSwitch, SwitchesResponse } from "@homeremote/types";
import { createSlice, createAsyncThunk, PayloadAction } from "@reduxjs/toolkit";
import fetchToJson from "../../../fetchToJson";
import ApiBaseState from "../../../Reducers/state.types";
import { logInfo } from "../LogCard/logSlice";
import { getSwitches } from "./getSwitchesThunk";

export const sendSwitchState = createAsyncThunk<
    void,
    { id: string; state: "on" | "off"; type: string }
>(
    `switchesList/sendSwitchState`,
    async ({ id, state, type }, { rejectWithValue, dispatch }) => {
        const json = await fetchToJson<SwitchesResponse>(
            `/api/switches/${id}`,
            {
                method: "POST",
                body: JSON.stringify({
                    state,
                    type,
                }),
            }
        );
        if (json.status !== "received") {
            throw new Error(`Can't set ${state}: ${json.status}`);
        } else {
            dispatch(logInfo(`Switch ${id} ${state}`));
        }
        dispatch(getSwitches());
    }
);

export interface SwitchBarListState extends ApiBaseState {
    switches: HomeRemoteSwitch[];
    expanded: string[];
}

export const initialState: SwitchBarListState = {
    switches: [],
    isLoading: false,
    error: false,
    expanded: [],
};

const switchBarListSlice = createSlice({
    name: "switchesList",
    initialState,
    reducers: {
        toggleExpandScene: (
            state,
            { payload: { sceneIdx } }: PayloadAction<{ sceneIdx: string }>
        ): void => {
            if (state.expanded.includes(sceneIdx)) {
                // Exists already, so toggle to "collapsed" and remove from the expanded array
                state.expanded = state.expanded.filter((id) => id !== sceneIdx);
            } else {
                // Does not exist yet, so toggle to "expanded" and add to the expanded array
                state.expanded.push(sceneIdx);
            }
        },
    },
    extraReducers: (builder) => {
        builder.addCase(getSwitches.pending, (draft, { payload }): void => {
            draft.error = initialState.error;
            draft.isLoading = true;
        });
        builder.addCase(getSwitches.fulfilled, (draft, { payload }): void => {
            draft.isLoading = initialState.isLoading;
            draft.switches = payload.switches ?? [];
        });
        builder.addCase(getSwitches.rejected, (draft, { error }): void => {
            draft.isLoading = initialState.isLoading;
            draft.switches = [];
            draft.error = error.message || "An error occurred";
        });
        builder.addCase(sendSwitchState.pending, (draft, { payload }): void => {
            draft.error = initialState.error;
            draft.isLoading = true;
        });
        builder.addCase(
            sendSwitchState.fulfilled,
            (draft, { payload }): void => {
                draft.isLoading = initialState.isLoading;
            }
        );
        builder.addCase(sendSwitchState.rejected, (draft, { error }): void => {
            draft.isLoading = initialState.isLoading;
            draft.error = error.message || "An error occurred";
        });
    },
});

export const { toggleExpandScene } = switchBarListSlice.actions;

export default switchBarListSlice.reducer;
