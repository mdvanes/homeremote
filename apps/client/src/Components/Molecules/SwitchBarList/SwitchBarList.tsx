import {
    DomoticzSendType,
    DomoticzType,
    HomeRemoteHaSwitch,
    HomeRemoteSwitch,
} from "@homeremote/types";
import { LinearProgress } from "@mui/material";
import { FC, Fragment, ReactElement, useEffect } from "react";
import { useSelector } from "react-redux";
import { RootState } from "../../../Reducers";
import { useAppDispatch } from "../../../store";
import { logError } from "../LogCard/logSlice";
import HaSwitchBar from "./HaSwitchBar";
import SwitchBar from "./SwitchBar";
import SwitchBarInnerButton from "./SwitchBarInnerButton";
import {
    DOMOTICZ_SELECTOR_STATES_OPTIONS,
    isDomoticzSelectorStateCode,
} from "./SwitchBarList.types";
import { getSwitches } from "./getSwitchesThunk";
import {
    SwitchBarListState,
    sendSwitchState,
    toggleExpandScene,
} from "./switchBarListSlice";

// Type is switchscene or switchlight
const getType = (type: DomoticzType): DomoticzSendType => {
    switch (type) {
        case "Group":
            return "switchscene";
        case "Scene":
            return "switchscene";
        case "Selector":
            return "selector";
        default:
            return "switchlight";
    }
};

const getLeftIcon = (label: string): string =>
    label === "Blinds" ? "arrow_drop_down" : "radio_button_checked";

const getRightIcon = (label: string): string =>
    label === "Blinds" ? "arrow_drop_up" : "radio_button_unchecked";

const getNameAndChildren = (
    name: string,
    children: HomeRemoteSwitch[] | false
): string => (children ? `${name}  \u25BE` : name);

const getLabel = (
    name: string,
    dimLevel: number | null,
    type: string
): string => {
    if (type === "selector" && isDomoticzSelectorStateCode(dimLevel)) {
        return `${name}: ${DOMOTICZ_SELECTOR_STATES_OPTIONS[dimLevel]}`;
    }
    return dimLevel !== null ? `${name} (${dimLevel}%)` : name;
};

type SendSomeState = (id: string, type: string) => void;

const isDomoticzSwitch = (
    dSwitch: HomeRemoteSwitch | HomeRemoteHaSwitch
): dSwitch is HomeRemoteSwitch => dSwitch.origin === "domoticz";

const isHaSwitch = (
    dSwitch: HomeRemoteSwitch | HomeRemoteHaSwitch
): dSwitch is HomeRemoteHaSwitch => dSwitch.origin === "home-assistant";

const mapSwitchToSwitchBar = (
    {
        idx,
        name,
        type,
        dimLevel,
        readOnly,
        status,
        children,
    }: HomeRemoteSwitch | HomeRemoteHaSwitch,
    sendOn: SendSomeState,
    sendOff: SendSomeState,
    labelAction: (() => void) | false
): ReactElement => (
    <SwitchBar
        key={`switch-${idx}`}
        icon={false}
        label={getLabel(
            getNameAndChildren(name, children ?? []),
            dimLevel,
            getType(type)
        )}
        labelAction={labelAction}
        leftButton={
            <SwitchBarInnerButton
                isReadOnly={readOnly}
                clickAction={(): void => sendOn(idx, getType(type))}
                icon={getLeftIcon(name)}
                isActive={status === "On"}
            />
        }
        rightButton={
            <SwitchBarInnerButton
                isReadOnly={readOnly}
                clickAction={(): void => sendOff(idx, getType(type))}
                icon={getRightIcon(name)}
                isActive={status === "Off"}
            />
        }
    />
);

const getLabelAction = (
    hasChildren: boolean,
    cbWithChildren: () => void,
    cbWithoutChildren: () => void
): (() => void) => (hasChildren ? cbWithChildren : cbWithoutChildren);

const SwitchBarList: FC = () => {
    const dispatch = useAppDispatch();
    const switches = useSelector<RootState, SwitchBarListState["switches"]>(
        (state: RootState) => state.switchesList.switches
    );
    const isLoading = useSelector<RootState, SwitchBarListState["isLoading"]>(
        (state: RootState) => state.switchesList.isLoading
    );
    const errorMessage = useSelector<RootState, SwitchBarListState["error"]>(
        (state: RootState) => state.switchesList.error
    );
    const expandedScenes = useSelector<
        RootState,
        SwitchBarListState["expanded"]
    >((state: RootState) => state.switchesList.expanded);

    const sendOn = (id: string, type: string): void => {
        dispatch(sendSwitchState({ id, type, state: "on" }));
    };

    const sendOff = (id: string, type: string): void => {
        dispatch(sendSwitchState({ id, type, state: "off" }));
    };

    useEffect(() => {
        dispatch(getSwitches());
    }, [dispatch]);
    useEffect(() => {
        dispatch(logError(errorMessage));
    }, [errorMessage, dispatch]);

    const switchBars = switches
        .filter(isDomoticzSwitch)
        .map((dSwitch: HomeRemoteSwitch) => {
            const hasChildren = Boolean(dSwitch.children);
            const labelAction = getLabelAction(
                hasChildren,
                () => {
                    dispatch(toggleExpandScene({ sceneIdx: dSwitch.idx }));
                    dispatch(getSwitches());
                },
                () => {
                    dispatch(getSwitches());
                }
            );
            const showChildren = expandedScenes.includes(dSwitch.idx);
            return (
                <Fragment key={`frag-${dSwitch.idx}`}>
                    {mapSwitchToSwitchBar(
                        dSwitch,
                        sendOn,
                        sendOff,
                        labelAction
                    )}
                    {hasChildren && showChildren ? (
                        <div style={{ padding: "0.5em" }}>
                            {dSwitch.children &&
                                dSwitch.children.map(
                                    (switchChild: HomeRemoteSwitch) =>
                                        mapSwitchToSwitchBar(
                                            switchChild,
                                            sendOn,
                                            sendOff,
                                            false
                                        )
                                )}
                        </div>
                    ) : null}
                </Fragment>
            );
        });

    const haSwitchBars = switches
        .filter(isHaSwitch)
        .map((haSwitch: HomeRemoteHaSwitch) => (
            <HaSwitchBar key={haSwitch.idx} haSwitch={haSwitch} />
        ));

    return (
        <Fragment>
            {switchBars}
            {haSwitchBars}
            {isLoading && <LinearProgress />}
        </Fragment>
    );
};

export default SwitchBarList;
