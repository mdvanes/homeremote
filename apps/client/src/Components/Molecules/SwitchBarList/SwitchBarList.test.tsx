import React from "react";
import * as ReactRedux from "react-redux";
import { render, fireEvent } from "@testing-library/react";
import SwitchBarList from "./SwitchBarList";
import { RootState } from "../../../Reducers";
import * as Slice from "./switchBarListSlice";
import SwitchBar from "./SwitchBar";
import { DomoticzTypeOptions } from "@homeremote/types";

type MockRootState = Pick<RootState, "switchesList">;

const mockRootState: MockRootState = {
    switchesList: {
        isLoading: false,
        error: false,
        switches: [
            {
                idx: "3",
                type: DomoticzTypeOptions.LightSwitch,
                name: "My Normal Light Switch",
                status: "On",
                dimLevel: null,
                readOnly: false,
                children: false,
            },
            {
                idx: "4",
                type: DomoticzTypeOptions.Group,
                name: "My Scene",
                status: "On",
                dimLevel: null,
                readOnly: false,
                children: [
                    {
                        idx: "5",
                        type: DomoticzTypeOptions.LightSwitch,
                        name: "My Nested Light Switch",
                        status: "On",
                        dimLevel: null,
                        readOnly: false,
                        children: false,
                    },
                ],
            },
            {
                idx: "6",
                type: DomoticzTypeOptions.Selector,
                name: "My Selector Switch",
                status: "FOO",
                dimLevel: 30,
                readOnly: true,
                children: false,
            },
            {
                idx: "7",
                type: DomoticzTypeOptions.LightSwitch,
                name: "My Dimmer",
                status: "FOO",
                dimLevel: 30,
                readOnly: false,
                children: false,
            },
            {
                idx: "8",
                type: DomoticzTypeOptions.LightSwitch,
                name: "Blinds",
                status: "FOO",
                dimLevel: null,
                readOnly: false,
                children: false,
            },
        ],
        expanded: [],
    },
};

// TODO refactor to replace all uses of this by renderSwitchBarList (see SwitchBarList_renderWith.test.tsx). Also remove all `jest.spyOn(ReactRedux, "useDispatch")`
const mockUseSelectorWith = ({ isLoading = false }): void => {
    jest.spyOn(ReactRedux, "useSelector").mockReset();
    jest.spyOn(ReactRedux, "useSelector").mockImplementation((fn) => {
        return fn({
            ...mockRootState,
            switchesList: { ...mockRootState.switchesList, isLoading },
        });
    });
};

describe("SwitchBarList", () => {
    const mockDispatch = jest.fn();

    beforeEach(() => {
        jest.spyOn(ReactRedux, "useDispatch").mockReturnValue(mockDispatch);
    });

    it("shows a normal light switch", () => {
        mockUseSelectorWith({});
        const { getByText, getByTestId } = render(<SwitchBarList />);
        expect(getByText("My Normal Light Switch")).toBeInTheDocument();
        // Test icons
        expect(
            getByTestId("switchbar-My Normal Light Switch")
        ).toMatchSnapshot();
    });

    it("shows a selector switch", () => {
        mockUseSelectorWith({});
        const { getByText, getByTestId } = render(<SwitchBarList />);
        expect(getByText(/My Selector Switch: armed/)).toBeInTheDocument();
        // Test read only state
        expect(
            getByTestId("switchbar-My Selector Switch: armed")
        ).toMatchSnapshot();
    });

    it("shows a dimmer", () => {
        mockUseSelectorWith({});
        const { getByText } = render(<SwitchBarList />);
        expect(getByText(/My Dimmer \(30%\)/)).toBeInTheDocument();
    });

    it("shows a blinds switch", () => {
        mockUseSelectorWith({});
        const { getByText, getByTestId } = render(<SwitchBarList />);
        expect(getByText("Blinds")).toBeInTheDocument();
        // Test icons
        expect(getByTestId("switchbar-Blinds")).toMatchSnapshot();
    });

    it("sends a switch state on clicking the 'off' button", () => {
        jest.spyOn(Slice, "sendSwitchState").mockReset();
        mockUseSelectorWith({});
        const { getByText, baseElement } = render(<SwitchBarList />);
        const offButton = baseElement.querySelector("button.makeStyles-root");
        expect(offButton).toBeInTheDocument();
        mockDispatch.mockReset();
        if (offButton) {
            fireEvent.click(offButton);
        }
        expect(getByText("My Normal Light Switch")).toBeInTheDocument();
        expect(Slice.sendSwitchState).toHaveBeenCalledWith({
            id: "3",
            state: "off",
            type: "switchlight",
        });
    });

    it("sends a switch state on clicking the 'on' button", () => {
        jest.spyOn(Slice, "sendSwitchState").mockReset();
        mockUseSelectorWith({});
        const { getByText, baseElement } = render(<SwitchBarList />);
        const onButton = baseElement.querySelector("button.makeStyles-active");
        expect(onButton).toBeInTheDocument();
        mockDispatch.mockReset();
        if (onButton) {
            fireEvent.click(onButton);
        }
        expect(getByText("My Normal Light Switch")).toBeInTheDocument();
        expect(Slice.sendSwitchState).toHaveBeenCalledWith({
            id: "3",
            state: "on",
            type: "switchlight",
        });
    });

    it("shows the loading state", () => {
        mockUseSelectorWith({ isLoading: true });
        const { baseElement } = render(<SwitchBarList />);
        expect(
            baseElement.querySelector(".MuiLinearProgress-root")
        ).toBeInTheDocument();
    });
});

describe("SwitchBar", () => {
    it("can show an icon", () => {
        const { baseElement } = render(
            <SwitchBar
                icon="foo"
                leftButton={<button>left</button>}
                rightButton={<button>right</button>}
                label="my bar"
                labelAction={false}
            />
        );
        expect(baseElement.querySelector(".MuiIcon-root")?.textContent).toBe(
            "foo"
        );
    });

    it("can have a text label", () => {
        const { baseElement } = render(
            <SwitchBar
                icon={false}
                leftButton={<button>left</button>}
                rightButton={<button>right</button>}
                label="my bar"
                labelAction={false}
            />
        );
        expect(
            baseElement.querySelector(".makeStyles-label")
        ).toHaveTextContent("my bar");
        expect(
            baseElement.querySelector(".makeStyles-label > button")
        ).not.toBeInTheDocument();
    });

    it("can have a button label", () => {
        const { baseElement } = render(
            <SwitchBar
                icon={false}
                leftButton={<button>left</button>}
                rightButton={<button>right</button>}
                label="my bar"
                labelAction={jest.fn()}
            />
        );
        expect(
            baseElement.querySelector(".makeStyles-label")
        ).toHaveTextContent("my bar");
        expect(
            baseElement.querySelector(".makeStyles-label > button")
        ).toBeInTheDocument();
    });
});
