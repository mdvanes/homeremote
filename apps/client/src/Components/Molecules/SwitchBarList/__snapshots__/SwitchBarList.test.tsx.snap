// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SwitchBarList shows a blinds switch 1`] = `
<div
  class="MuiCardContent-root css-bi9i8-MuiCardContent-root-buttonGroup"
  data-testid="switchbar-Blinds"
>
  <button
    class="css-l44rl-root"
  >
    <span
      aria-hidden="true"
      class="material-icons notranslate MuiIcon-root MuiIcon-fontSizeMedium material-icons css-kp9ftd-MuiIcon-root"
    >
      arrow_drop_down
    </span>
  </button>
  <span
    class="css-ccz72j-label"
  >
    <i />
    <p
      class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
    >
      Blinds
    </p>
  </span>
  <button
    class="css-l44rl-root"
  >
    <span
      aria-hidden="true"
      class="material-icons notranslate MuiIcon-root MuiIcon-fontSizeMedium material-icons css-kp9ftd-MuiIcon-root"
    >
      arrow_drop_up
    </span>
  </button>
</div>
`;

exports[`SwitchBarList shows a normal light switch 1`] = `
<div
  class="MuiCardContent-root css-bi9i8-MuiCardContent-root-buttonGroup"
  data-testid="switchbar-My Normal Light Switch"
>
  <button
    class="css-nm7ep5-active"
  >
    <span
      aria-hidden="true"
      class="material-icons notranslate MuiIcon-root MuiIcon-fontSizeMedium material-icons css-kp9ftd-MuiIcon-root"
    >
      radio_button_checked
    </span>
  </button>
  <span
    class="css-ccz72j-label"
  >
    <i />
    <p
      class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
    >
      My Normal Light Switch
    </p>
  </span>
  <button
    class="css-l44rl-root"
  >
    <span
      aria-hidden="true"
      class="material-icons notranslate MuiIcon-root MuiIcon-fontSizeMedium material-icons css-kp9ftd-MuiIcon-root"
    >
      radio_button_unchecked
    </span>
  </button>
</div>
`;

exports[`SwitchBarList shows a selector switch 1`] = `
<div
  class="MuiCardContent-root css-bi9i8-MuiCardContent-root-buttonGroup"
  data-testid="switchbar-My Selector Switch: armed"
>
  <button
    class="css-l44rl-root"
  >
    <div
      class="dummy"
    />
  </button>
  <span
    class="css-ccz72j-label"
  >
    <i />
    <p
      class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
    >
      My Selector Switch: armed
    </p>
  </span>
  <button
    class="css-l44rl-root"
  >
    <div
      class="dummy"
    />
  </button>
</div>
`;
